token/services/db/sql/identity.go

```diff
@@ -67,37 +67,49 @@ func NewIdentityDB(db *sql.DB, tablePrefix string, createSchema bool, signerInfo
 }
 
 func (db *IdentityDB) AddConfiguration(wp driver.IdentityConfiguration) error {
-	query := fmt.Sprintf("INSERT INTO %s (id, type, url) VALUES ($1, $2, $3)", db.table.IdentityConfigurations)
+	query := fmt.Sprintf("INSERT INTO %s (id, type, url, conf, raw) VALUES ($1, $2, $3, $4, $5)", db.table.IdentityConfigurations)
 	logger.Debug(query)
 
-	_, err := db.db.Exec(query, wp.ID, wp.Type, wp.URL)
+	_, err := db.db.Exec(query, wp.ID, wp.Type, wp.URL, wp.Config, wp.Raw)
 	return err
 }
 
 func (db *IdentityDB) IteratorConfigurations(configurationType string) (driver.Iterator[driver.IdentityConfiguration], error) {
-	query := fmt.Sprintf("SELECT id, url FROM %s WHERE type = $1", db.table.IdentityConfigurations)
+	query := fmt.Sprintf("SELECT id, url, conf, raw FROM %s WHERE type = $1", db.table.IdentityConfigurations)
 	logger.Debug(query)
 	rows, err := db.db.Query(query, configurationType)
 	if err != nil {
 		return nil, err
 	}
-	return &WalletPathStorageIterator{rows: rows, configurationType: configurationType}, nil
+	return &IdentityConfigurationIterator{rows: rows, configurationType: configurationType}, nil
 }
 
-func (db *IdentityDB) StoreAuditInfo(id, info []byte) error {
-	query := fmt.Sprintf("INSERT INTO %s (identity_hash, identity, info) VALUES ($1, $2, $3)", db.table.AuditInfo)
+func (db *IdentityDB) ConfigurationExists(id, typ string) (bool, error) {
+	result, err := QueryUnique[string](db.db,
+		fmt.Sprintf("SELECT id FROM %s WHERE id=$1 AND type=$2", db.table.IdentityConfigurations),
+		id, typ,
+	)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed getting configuration for [%s:%s]", id, typ)
+	}
+	logger.Debugf("found configuration for [%s:%s]", id, typ)
+	return len(result) != 0, nil
+}
+
+func (db *IdentityDB) StoreIdentityData(id []byte, identityAudit []byte, tokenMetadata []byte, tokenMetadataAudit []byte) error {
+	query := fmt.Sprintf("INSERT INTO %s (identity_hash, identity, identityAuditInfo, tokenMetadata, tokenMetadataAuditInfo) VALUES ($1, $2, $3, $4, $5)", db.table.IdentityInfo)
```

Comment:
the naming of the columns is not consistent

Reply:
👍